Rivet/Functions/Checkpoint-Migration.ps1

```diff
@@ -0,0 +1,58 @@
+
+function Checkpoint-Migration
+{
+    <#
+    .SYNOPSIS
+    Checkpoints the current state of the database so that it can be re-created.
+
+    .DESCRIPTION
+    The `Checkpoint-Migration` function pushes all migrations to the current database. The database is then exported to
+    a schema.ps1 script which can be used to re-create the current state of the database.
+
+    .EXAMPLE
+    Checkpoint-Migration -Database $Database -Environment $Environment -ConfigFilePath $ConfigFilePath
+
+    Demonstrates how to checkpoint a migration.
+    #>
+    [CmdletBinding()]
+    param(
+        # The database(s) to migrate.
+        [Parameter(Mandatory)]
+        [String[]] $Database,
+
+        # The environment you're working in.
+        [String] $Environment,
+
+        # The path to the Rivet configuration file. Default behavior is to look in the current directory for a `rivet.json` file.
+        [String] $ConfigFilePath
+    )
+
+    Set-StrictMode -Version 'Latest'
+    Use-CallerPreference -Cmdlet $PSCmdlet -SessionState $ExecutionContext.SessionState
+
+    try
+    {
+        $checkpointedMigrations = [System.Collections.ArrayList]::new()
+        [Rivet.Configuration.Configuration]$settings = Get-RivetConfig -Database $Database -Path $ConfigFilePath -Environment $Environment
+        Invoke-Rivet -Push @PSBoundParameters | Out-Null
+
+        foreach( $databaseItem in $settings.Databases )
+        {
+            $migration = Export-Migration -SqlServerName $settings.SqlServerName -Database $databaseItem.Name
+            $migration = $migration -join [Environment]::NewLine
+            $checkpointedMigrations.Add($migration) | Out-Null
+        }
+
+        return $checkpointedMigrations
+    }
+    catch
+    {
+        $firstException = $_.Exception
+        while( $firstException.InnerException )
+        {
+            $firstException = $firstException.InnerException
+        }
+    
+        Write-Error ('{0} database migration failed: {1}.' -f $Database,$firstException.Message)

```

Comment:
Do we need this try/catch with this error message? I think we can just let `Invoke-Rivet` write the error if it fails.